File: csharp/grade-school/GradeSchool.cs
using System.Collections.Generic;

public class School
{
  public Dictionary<int, List<string>> Roster { get; private set; }

  public School()
  {
    Roster = new Dictionary<int, List<string>>();
  }

  public void Add(string student, int grade)
  {
    if (Roster.ContainsKey(grade))
    {
      Roster[grade].Add(student);
    }
    else
    { 
      Roster[grade] = new List<string>() { student };
    }
    Roster[grade].Sort();
  }

  public List<string> Grade(int grade)
  {
    return Roster.ContainsKey(grade) ? Roster[grade] : new List<string>();
  }
}

File: javascript/grade-school/grade-school.js
function School() {
  var roll = {};
  this.roster = function() {
    return roll;
  };
  this.add = function(student, grade) {
    if (roll[grade] === undefined) roll[grade] = [student];
    else roll[grade].push(student);
    roll[grade].sort();
  };
  this.grade = function(number) {
    return roll[number] === undefined ? [] : roll[number];
  }
};

module.exports = School;
